libs/entitlement/asm-provider/src/test/java/org/elasticsearch/entitlement/instrumentation/impl/InstrumentationServiceImplTests.java

```diff
@@ -124,15 +136,136 @@ public void testInstrumentationTargetLookupWithOverloads() throws IOException, C
         assertThat(
             methodsMap,
             hasEntry(
-                equalTo(new MethodKey("org/example/TestTargetClass", "someOverloadedMethod", List.of("I", "I"), true)),
+                equalTo(new MethodKey("org/example/TestTargetClass", "staticMethodWithOverload", List.of("I", "I"), true)),
                 equalTo(
                     new CheckerMethod(
                         "org/elasticsearch/entitlement/instrumentation/impl/InstrumentationServiceImplTests$TestCheckerOverloads",
-                        "checkInstanceMethodWithOverload",
+                        "check$org_example_TestTargetClass$staticMethodWithOverload",
                         List.of("Ljava/lang/Class;", "I", "I")
                     )
                 )
             )
         );
     }
+
+    public void testParseCheckerMethodSignatureStaticMethod() {
+        var methodKey = InstrumentationServiceImpl.parseCheckerMethodSignature(
+            "check$org_example_TestClass$staticMethod",
+            new Type[] { Type.getType(Class.class) }
+        );
+
+        assertTrue(methodKey.isStatic());
+        assertThat(methodKey.methodName(), equalTo("staticMethod"));
+        assertThat(methodKey.className(), equalTo("org/example/TestClass"));
+        assertThat(methodKey.parameterTypes(), empty());
+    }
+
+    public void testParseCheckerMethodSignatureStaticMethodWithArgs() {
+        var methodKey = InstrumentationServiceImpl.parseCheckerMethodSignature(
+            "check$org_example_TestClass$staticMethod",
+            new Type[] { Type.getType(Class.class), Type.getType("I"), Type.getType(String.class) }
+        );
+
+        assertTrue(methodKey.isStatic());
+        assertThat(methodKey.methodName(), equalTo("staticMethod"));
+        assertThat(methodKey.className(), equalTo("org/example/TestClass"));
+        assertThat(methodKey.parameterTypes(), contains("I", "java/lang/String"));
+    }
+
+    public void testParseCheckerMethodSignatureStaticMethodInnerClass() {
+        var methodKey = InstrumentationServiceImpl.parseCheckerMethodSignature(
+            "check$org_example_TestClass$InnerClass$staticMethod",
+            new Type[] { Type.getType(Class.class) }
+        );
+
+        assertTrue(methodKey.isStatic());
+        assertThat(methodKey.methodName(), equalTo("staticMethod"));
+        assertThat(methodKey.className(), equalTo("org/example/TestClass$InnerClass"));
+        assertThat(methodKey.parameterTypes(), empty());
```

Comment:
We require that every field in the resulting `MethodKey` has the exact right value. Checking each field individually means we'd need to remember to fix these tests if we ever add more fields.

If it's rewritten in this style, then we'll get a helpful compile error when new fields are added:

```
assertEquals(new MethodKey(...), methodKey);
```

Reply:
(Oh also... we probably want to get rid of `isStatic` at some point. I forget why I added that, but it's not necessary and it's redundant.)

Reply:
> If it's rewritten in this style, then we'll get a helpful compile error when new fields are added

I like the idea, and the reasoning behind it. Will rewrite the assertions this way.